PxtlCa.XmlCommentMarkDownGenerator.MSBuild/GenerateMarkdown.cs

```diff
@@ -0,0 +1,273 @@
+        /// Both existing markdown files and the generated files are merged.
+        /// </summary>
+        public bool MergeFiles { get; set; }
+
+        /// <summary>
+        /// The file to be created by the merge.  Unused if MergeFiles evaluates to false.
+        /// </summary>
+        public ITaskItem OutputFile { get; set; }
+
+        /// <summary>
+        /// Defaults to false. When true unexpected tags in the documentation
+        /// will generate warnings rather than errors. 
+        /// </summary>
+        public bool WarnOnUnexpectedTag { get; set; } = false;
+
+      
+
+        /// <summary>
+        /// Runs the task as configured
+        /// </summary>
+        /// <returns>true if task has succeeded</returns>
+        public override bool Execute()
+        {
+            if (InputXml.Length == 0)
+            {
+                Log.LogError("InputXml cannot be empty");
+            }
+            else
+            {
+                UpdateParametersFromInput();
+                if (MergeFiles && (null == OutputFile))
+                {
+                    Log.LogError("OutputFile must be specified if input files are merged");
+                }
+                else if (DocumentationPathIsFile && InputXml.Length != 1)
+                {
+                    Log.LogError("DocumentationPath must specify a directory if more than one input XML value is supplied");
+                }
+                else
+                {
+                    try
+                    {
+                        CreateDirectoryIfNeeded();
+                        GenerateFiles();
+                        if (MergeFiles)
+                        {
+                            Merge();
+                        }
+                        return true;
+                    }
+                    catch (Exception ex)
+                    {
+                        LoggedException = ex;
+                        Log.LogErrorFromException(ex);
+                    }
+                }
+            }
+            return false;
+        }
+
+        /// <summary>
+        /// This updates the tag and merge parameters based on the input front matter
+        /// </summary>
+        public void UpdateParametersFromInput()
+        {
+            if(DocumentationPathIsFile)
+            {
+                if (TryGetFrontMatter(DocumentationPath.ItemSpec, out string frontMatter, out bool isEmpty) && 
+                    (!isEmpty))
+                {
+                    ReadOptionsFromString(frontMatter);
+                    return;
+                }
+            }
+            else
+            {
+                var mdFiles = Directory.EnumerateFiles(DocumentationPath.ItemSpec, "*.md", SearchOption.AllDirectories).ToList();
+                foreach (var mdFile in mdFiles)
+                {
+                    if (TryGetFrontMatter(mdFile, out string frontMatter, out bool isEmpty) &&
```

Comment:
I'm not fond of the fact that it does this in apparently undefined order isntead of looking for an .md file that matches the XML file by name first, or look fora  file named readme.md first... but either way, it's good for now.  This looks like an opportunity for future enhancement.